### PR TITLE
Zustand store

### PR DESCRIPTION
**Pull request by OlegGre4ka**
- [ ] - connected and configured Zustand state manager instead Mobx, it allows us to use the hook approach instead of the observer pattern

### CODE REVIEW

File: web-client/src/App.tsx [hunk 1]
```diff
@@ -1,13 +1,22 @@
 import { BrowserRouter } from 'react-router-dom';
 import MainLayout from 'components/MainLayout';
 import AppRoutes from 'routes/index';
+import {IntlProvider} from 'react-intl';
+import messages from 'intl/index';
+import  {useSettingsStore}  from 'store/useSettings';
+
+const App = () => {
+const {language} = useSettingsStore();
```
Comment (valoricDe) [inline web-client/src/App.tsx:9]: do you have eslint enabled (and prettier added) in intellj or visual studio code?

File: web-client/src/components/LanguageSelect/index.tsx [hunk 2]
```diff
@@ -1,29 +1,30 @@
...
 const LangSelect: React.FC<LangSelectProps> = ({ style }) =>{
-    const [lang, setLang] = useState("ua");
-    const handleChange = (event: SelectChangeEvent) => {
-        setLang(event.target.value as string);
-    };
+    const { isMobile, language, setLanguage } = useSettingsStore();
+
     return(
-        <Box sx={{ width: '100px', height: '40px' }}>
+        <Box sx={{width: isMobile ? '100px' : '140px', height: '40px'}}>
```
Comment (valoricDe) [inline web-client/src/components/LanguageSelect/index.tsx:16]: is it possible to use something like this: https://mui.com/material-ui/customization/breakpoints/?

Comment (valoricDe) [reply web-client/src/components/LanguageSelect/index.tsx:16]: To utilize the framework and react consitently then we would not need the isMobile state in useSettingsStore

File: web-client/src/components/MainLayout/index.tsx [hunk 3]
```diff
@@ -70,7 +68,7 @@ const MainLayout = observer(({ children }: IMainLayout) => {
                     <AppBar component="nav" style={{ backgroundColor: 'var(--white-color)' }}>
                         <Toolbar style={{ display: 'flex', justifyContent: 'space-between', alignItems: 'center' }}>
                             <NavLinkLogo to="/"><img src={Logo} alt="Logo" style={{ marginTop: '12px' }} /></NavLinkLogo>
-                            <FlexBox justify="space-between" width="360px">
+                            <FlexBox justify="space-between" width={language==='en' ? "360px" : "600px"}>
```
Comment (valoricDe) [inline web-client/src/components/MainLayout/index.tsx:71]: why you have different width based on the language en?
Will these numbers like 100px vs 140 and 360px vs 600px be reused somewhere?

File: web-client/src/App.tsx (hunk 1 above)
Comment (OlegGre4ka) [reply web-client/src/App.tsx:9]: Yes

File: web-client/src/components/LanguageSelect/index.tsx (hunk 2 above)
Comment (OlegGre4ka) [reply web-client/src/components/LanguageSelect/index.tsx:16]: 1.I have changed materialUI Box to custom component FlexBox;
2.I have deleted isMobile from the state in useSettingsStore and created custom hook useMobileSize

File: web-client/src/components/MainLayout/index.tsx (hunk 3 above)
Comment (OlegGre4ka) [reply web-client/src/components/MainLayout/index.tsx:71]: 1.In Ukrainian, these words are longer, and this block needs a different width for better positioning
2.No, only in these places

Comment (valoricDe) [reply web-client/src/components/MainLayout/index.tsx]: Ah alright 👍

**Review by valoricDe — APPROVED** [review]

**Pull request merged**